pkg/image/importer/client_test.go

```diff
@@ -226,6 +226,40 @@ func TestPing(t *testing.T) {
 	}
 }
 
+func TestPingRecordRealm(t *testing.T) {
+	retriever := NewContext(http.DefaultTransport, http.DefaultTransport).WithCredentials(NewCredentialsForSecrets([]kapi.Secret{})).(*repositoryRetriever)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Docker-Distribution-API-Version", "registry/2.0")
+		w.Header().Set("WWW-Authenticate", `Bearer realm="https://auth.example.com/token"`)
+		w.WriteHeader(http.StatusUnauthorized)
+	}))
+	uri, _ := url.Parse(server.URL)
+
+	_, err := retriever.ping(*uri, true, retriever.context.InsecureTransport)
+	if err != nil {
+		t.Errorf("Unexpected error, got %v", err)
+	}
+	scs, ok := retriever.credentials.(*SecretCredentialStore)
+	if !ok {
+		t.Fatalf("Unexpected credential store type %T", retriever.credentials)
+	}
+	if len(scs.realmStore.List()) != 1 {
+		t.Errorf("Unexpected realm # entries, expected 1, got %d", len(scs.realmStore.List()))
+	}
+	obj := scs.realmStore.List()[0]
+	ru, ok := obj.(*realmURL)
+	if !ok {
+		t.Errorf("Unexpected object type, expected realmURL, got %T", obj)
+	}
+	if ru.realm.String() != "https://auth.example.com/token" {
```

Comment:
*nit:* a good candidate for `const`

Reply:
I didn't bother since this is just test.